aspnetcore/blazor/security/blazor-web-app-with-entra.md

```diff
@@ -0,0 +1,172 @@
+* The app uses [Microsoft Entra ID](https://www.microsoft.com/security/business/microsoft-entra), based on [Microsoft Identity Web](/entra/msal/dotnet/microsoft-identity-web/) packages.
+* Automatic non-interactive token refresh is managed by the framework.
+* The app uses server-side and client-side service abstractions to display generated weather data:
+  * When rendering the `Weather` component on the server to display weather data, the component uses the `ServerWeatherForecaster` on the server to directly obtain weather data (not via a web API call).
+  * When the `Weather` component is rendered on the client, the component uses the `ClientWeatherForecaster` service implementation, which uses a preconfigured <xref:System.Net.Http.HttpClient> (in the client project's `Program` file) to make a web API call to the server project's Minimal API (`/weather-forecast`) for weather data. The Minimal API endpoint obtains the weather data from the `ServerWeatherForecaster` class and returns it to the client for rendering by the component.
+
+## Sample app
+
+The sample app consists of two projects:
+
+* `BlazorWebAppEntra`: Server-side project of the Blazor Web App, containing an example [Minimal API](xref:fundamentals/minimal-apis) endpoint for weather data.
+* `BlazorWebAppEntra.Client`: Client-side project of the Blazor Web App.
+
+Access sample apps through the latest version folder from the repository's root with the following link. The projects are in the `BlazorWebAppEntra` folder for .NET 9 or later.
+
+[View or download sample code](https://github.com/dotnet/blazor-samples) ([how to download](xref:blazor/fundamentals/index#sample-apps))
+
+## Server-side Blazor Web App project (`BlazorWebAppEntra`)
+
+The `BlazorWebAppEntra` project is the server-side project of the Blazor Web App.
+
+Authentication state provider services are provided by the framework with a call to `AddAuthenticationStateSerialization` on the Razor components builder in the `Program` file, which captures the user's authentication state and flows it from the server to the client.
+
+The `BlazorWebAppEntra.http` file can be used for testing the weather data request. Note that the `BlazorWebAppEntra` project must be running to test the endpoint, and the endpoint is hardcoded into the file. For more information, see <xref:test/http-files>.
+
+> [!NOTE]
+> Internally, the server project uses <xref:Microsoft.AspNetCore.Http.IHttpContextAccessor>/<xref:Microsoft.AspNetCore.Http.HttpContext>, but never for interactively-rendered components. For more information, see <xref:blazor/security/server/interactive-server-side-rendering#ihttpcontextaccessorhttpcontext-in-razor-components>.
```

Comment:
Did you make a BFF update to the sample that uses this? I used this in the `BlazorWebAppOidcBff` sample, but not the Entra sample (at least not yet).

I saw https://github.com/dotnet/blazor-samples/pull/360, but that doesn't need the `IHttpContextAccsesor` since it doesn't proxy the request and forward the access token to a separate backend process.

Reply:
My bad on that ... I thought that the auth state services that are now part of the framework used it. I'll remove this language.

WRT the sample app updates for the new BWA+Entra sample, I didn't add `IHttpContextAccessor`/`HttpContext` outside of the client-side `HttpClient` for the weather data.